force-app/main/domain/HouseholdNamingUserControlledFields.cls

```diff
@@ -41,19 +41,30 @@ public inherited sharing class HouseholdNamingUserControlledFields {
     private SObject oldRecord;
 
     public HouseholdNamingUserControlledFields(String str) {
-        this.value = str == null ? ';' : str + ';';
+        // Ensure that double semicolons are not allowed when defining the user controlled naming fields.
+        // This will only add a new semicolon if the previous string does not already end with one.
+        this.value = str == null ? ';' : (str.endsWith(';') ? str : str + ';');
```

Comment:
Was this issue (double-semicolon) one of the root issues of the problems being addressed or was this just an observation?